mlflow/tensorflow.py

```diff
@@ -13,76 +13,213 @@
 from __future__ import absolute_import
 
 import os
+import shutil
 
 import pandas
 import tensorflow as tf
 
+import mlflow
 from mlflow import pyfunc
+from mlflow.exceptions import MlflowException
 from mlflow.models import Model
-from mlflow.tracking.fluent import _get_or_start_run, log_artifacts
+from mlflow.protos.databricks_pb2 import DIRECTORY_NOT_EMPTY
+from mlflow.tracking.utils import _get_model_log_dir
+from mlflow.utils.file_utils import _copy_file_or_tree
 
+FLAVOR_NAME = "tensorflow"
 
-class _TFWrapper(object):
+
+def log_model(tf_saved_model_dir, tf_meta_graph_tags, tf_signature_def_key, artifact_path,
```

Comment:
I would be more on board with the idea of specifying default values if we had reasonable defaults for both parameters. It seems that the user will need to know the value of their `SavedModel`'s `signature_def_key` because a default value for this parameter is somewhat difficult to nail down. Finding this value requires the same understanding and skillset as finding the value for the meta graph tags. Additionally, there are many Tensorflow examples that save models with the `tf_meta_graph_tags = tag_constants.TRAINING`. 

For these reasons, I'm more in favor of requiring explicit values for both parameters. If you still feel strongly about having a default value for `tf_meta_graph_tags`, we can incorporate it.